docs/docs/data-structures/tree/schema-evolution/types-of-changes.mdx

```diff
@@ -0,0 +1,213 @@
+---
+title: Types of Schema Changes
+sidebar_position: 2
+---
+
+# Types of Schema Changes
+
+Understanding which types of schema changes are safe versus breaking is crucial for evolving your [SharedTree](../index.mdx) schemas without disrupting existing applications.
+
+## Compatible Changes
+
+The following changes are [backwards compatible](index.mdx#backwards-compatibility).
+However, it's important that applications carefully whether the change is [forwards compatible](./index.mdx#forwards-compatibility) inherently or if a [staged rollout](./index.mdx#staged-rollouts) is necessary.
+
+### Adding Optional Fields
+
+Adding optional fields to existing node types is always [backwards compatible](./index.mdx#understanding-compatibility):
+
+```typescript
+// Before
+class Note extends factory.object("Note", {
+    id: factory.string,
+    text: factory.string,
+}) {}
+
+// After - Safe addition
+class Note extends factory.object("Note", {
+    id: factory.string,
+    text: factory.string,
+    color: factory.optional(factory.string), // New optional field
+}) {}
+```
+
+However, older clients that don't know about this new field will not be able to read it.
+To ensure [forwards compatibility](./index.mdx#understanding-compatibility), use [`allowUnknownOptionalFields`](../../../api/fluid-framework/objectnodeschema-interface#allowunknownoptionalfields-property) to handle new optional fields added by newer clients:
```

Comment:
Emphasize that this needs to _already_ be set on the old clients before the field is added. If it's not already set, then this degenerates into an un-forwards compatible change and requires a staged rollout - because the app has to stage the enabling of `allowUnknownOptionalFields`.